src/main/java/com/scalyr/log4j/ScalyrAppender.java

```diff
@@ -2,41 +2,64 @@
 
 import com.scalyr.api.logs.EventAttributes;
 import com.scalyr.api.logs.Events;
+import com.scalyr.util.Util;
 import org.apache.log4j.AppenderSkeleton;
 import org.apache.log4j.Level;
 import org.apache.log4j.spi.LoggingEvent;
 
+import java.util.Map;
+
 public class ScalyrAppender extends AppenderSkeleton {
     private String apiKey;
     private String serverHost = "";
+    private String env = "";
     private String logfile = "log4j";
     private String parser = "log4j";
+    private String extraAttributes = "[]";
     private Integer maxBufferRam;
 
-    public String getServerHost() {
-        return this.serverHost == null ? "" : this.serverHost.trim();
-    }
+    public String getServerHost() { return this.serverHost == null ? "" : this.serverHost.trim(); }
 
-    public void setServerHost(String serverHost) {
-        this.serverHost = serverHost;
-    }
+    public void setServerHost(String serverHost) { this.serverHost = serverHost; }
 
-    public String getLogfile() {
-        return logfile;
-    }
+    public String getLogfile() { return logfile; }
 
-    public void setLogfile(String logfile) {
-        this.logfile = logfile;
-    }
+    public void setLogfile(String logfile) { this.logfile = logfile; }
 
-    public String getParser() {
-        return parser;
-    }
+    public String getParser() { return parser; }
 
-    public void setParser(String parser) {
-        this.parser = parser;
-    }
+    public void setParser(String parser) { this.parser = parser; }
+
+    public void setEnv(String env) { this.env = env; }
+
+    public String getEnv() { return env; }
 
+    /**
+     * Use this to describe any additional server attributes. Takes a string that is kv pairs separated by a comma e.g.:
+     * <pre>
+     *   appName=appofdoom,key1=value1
+     * </pre>
+     * An example of what to put in your logback.xml:
+     <pre><code>
+     &lt;configuration&gt;
+       &lt;appender name="scalyr" class="com.scalyr.logback.ScalyrAppender"&gt;
+         &lt;apiKey&gt;YOUR KEY&lt;/apiKey&gt;
+         &lt;extraAttributes&gt;appName=appofdoom,zodiac=rooster&lt;/extraAttributes&gt;
+         ...
+       &lt;/appender&gt;
+     &lt;/configuration&gt;
+     </code></pre>
+     * <p>
+     * The extraAttributes will be appended to the serverAttributes list and will be searchable in scalyr by a query like:
+     * <pre>
+     *   $zodiac == "rooster" "hello world"
+     * </pre>
+     *
+     * @param extraAttributes String that is a JSON representation of additional key/value pairs to add.
```

Comment:
This is not actually JSON... how about "An optional list of comma-separated key=value pairs to add to serverAttributes"?

Reply:
👍